modules/react-native-wallet/android/src/main/java/com/wallet/WalletModule.kt

```diff
@@ -0,0 +1,304 @@
+package com.wallet
+
+import android.R.attr.data
+import android.app.Activity
+import android.app.Activity.RESULT_CANCELED
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.util.Log
+import com.facebook.react.bridge.ActivityEventListener
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.PromiseImpl
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.google.android.gms.tapandpay.TapAndPay
+import com.google.android.gms.tapandpay.TapAndPayClient
+import com.google.android.gms.tapandpay.issuer.PushTokenizeRequest
+import com.wallet.Utils.getAsyncResult
+import com.wallet.Utils.toCardData
+import com.wallet.event.OnCardActivatedEvent
+import com.wallet.model.CardStatus
+import com.wallet.model.WalletData
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.nio.charset.Charset
+import java.util.Locale
+
+
+class WalletModule internal constructor(context: ReactApplicationContext) : WalletSpec(context) {
+
+  companion object {
+    const val NAME = "Wallet"
+    const val REQUEST_CODE_PUSH_TOKENIZE: Int = 3
+    const val REQUEST_CREATE_WALLET: Int = 4
+
+    const val TSP_VISA: String = "VISA"
+    const val TSP_MASTERCARD: String = "MASTERCARD"
+  }
+
+  private var tapAndPayClient: TapAndPayClient? = null
+  private var pendingCreateWalletPromise: Promise? = null
+  private var pendingAddCardPromise: Promise? = null
+
+
+  init {
+    reactApplicationContext.addActivityEventListener(object : ActivityEventListener {
+      override fun onActivityResult(
+        activity: Activity?, requestCode: Int, resultCode: Int, intent: Intent?
+      ) {
+        if (requestCode == REQUEST_CREATE_WALLET) {
+          pendingCreateWalletPromise?.let {
+            if (resultCode == RESULT_OK) {
+              it.resolve(true);
+              return
+            }
+            it.resolve(false);
```

Comment:
```suggestion
            it.resolve(resultCode == RESULT_OK);
```

I'm not sure how did `return` behave here - did it return out of the `let` lamba or out of the `onActivityResult` method?